examples/spouts/go-rabbitmq-spout/source/main.go

```diff
@@ -0,0 +1,408 @@
+			if latestClosedCommit != "" {
+				log.Printf("switching branch %s HEAD to %s", opts.switchBranch, latestClosedCommit)
+				err := pc.CreateBranch(opts.repoName, opts.switchBranch, latestClosedCommit, nil)
+				if err != nil {
+					errors <- err
+					return
+				}
+			}
+		}
+	}
+}
+
+func consume(pc *client.APIClient, msgs <-chan amqp.Delivery, opts *options, commits chan<- string, errors chan<- error) {
+
+	// Messages to write to file.
+	msgC := make(chan amqp.Delivery, opts.prefetch)
+
+	// Simple goroutine to read incoming messages from a channel
+	go bufferReceive(pc, msgC, opts, commits, errors)
+
+	// This may look funny, but the msgs channel is unbuffered. I don't like debugging channel deadlocks, do you?
+	for msg := range msgs {
+		// Write message to the buffer
+		msgC <- msg
+
+	}
+
+}
+
+type options struct {
+	prefetch       int
+	flushInterval  int
+	switchInterval int
+	topic          string
+	err            error
+	test           bool
+	overwrite      bool
+	repoName       string
+	commitBranch   string
+	switchBranch   string
+	ext            string
+}
+
+func parseOptions(opts *options) {
+	var err error
+
+	// Prefetch
+	prefetchStr, ok := os.LookupEnv("PREFETCH")
+	if ok {
+		opts.prefetch, err = strconv.Atoi(prefetchStr)
+		if err != nil {
+			log.Fatal("unable to parse PREFETCH")
+		}
+	}
+
+	// Flush Interval
+	flushIntStr, ok := os.LookupEnv("FLUSH_INTERVAL_MS")
+	if ok {
+		opts.flushInterval, err = strconv.Atoi(flushIntStr)
+		if err != nil {
+			log.Fatal("unable to parse FLUSH_INTERVAL_MS")
+		}
+	}
+
+	// Branch Switch Interval
+	switchIntStr, ok := os.LookupEnv("SWITCH_INTERVAL_MS")
+	if ok {
+		opts.switchInterval, err = strconv.Atoi(switchIntStr)
+		if err != nil {
+			log.Fatal("unable to parse SWITCH_INTERVAL_MS")
+		}
+	}
+
+	// File Extension
+	extStr, ok := os.LookupEnv("EXTENSION")
+	if ok {
+		opts.ext = extStr
+	}
+
+	testStr, ok := os.LookupEnv("TEST")
+	if ok {
+		opts.test, err = strconv.ParseBool(testStr)
+		if err != nil {
+			log.Fatal("unable to parse TEST")
+		}
+	}
+
+	switchBranchName, ok := os.LookupEnv("SWITCH_BRANCH")
+	if ok {
+		opts.switchBranch = switchBranchName
+	}
+
+	commitBranchName, ok := os.LookupEnv("COMMIT_BRANCH")
+	if ok {
+		opts.commitBranch = commitBranchName
+	}
+
+	// Topic/Queue for this to read from
+	opts.topic = os.Getenv("TOPIC")
+
+	opts.repoName = os.Getenv("PPS_PIPELINE_NAME")
+}
+
+func dialRabbit() (*amqp.Connection, error) {
+	rabbitURL := os.Getenv("RABBITMQ_HOST")
+	if rabbitURL == "" {
+		return nil, fmt.Errorf("RABBITMQ_HOST environment variable not set")
+	}
+
+	rabbitUser := os.Getenv("RABBITMQ_USER")
+	if rabbitUser == "" {
+		return nil, fmt.Errorf("RABBITMQ_USER environment variable not set")
+	}
+	rabbitPassword := os.Getenv("RABBITMQ_PASSWORD")
+	if rabbitPassword == "" {
+		return nil, fmt.Errorf("RABBITMQ_PASSWORD environment variable not set")
+	}
+
+	return amqp.Dial("amqp://" + rabbitUser + ":" + rabbitPassword + "@" + rabbitURL)
```

Comment:
Hey Daniel,

I set up a AWS MQ - RabbitMQ to connect from Pach Hub. It worked like a charm but only supports amqps. Added it to the code. It felt a little too much to fork your repo and push a PR in your PR branch. ;-) Feel free to patch/adapt directly or I can do it after the merge. LGTM

```

diff --git a/examples/spouts/go-rabbitmq-spout/pipelines/spout.pipeline.json b/examples/spouts/go-rabbitmq-spout/pipelines/spout.pipeline.json
index 7549aee50..12e5ca5ae 100644
--- a/examples/spouts/go-rabbitmq-spout/pipelines/spout.pipeline.json
+++ b/examples/spouts/go-rabbitmq-spout/pipelines/spout.pipeline.json
@@ -17,6 +17,7 @@
             "SWITCH_INTERVAL_MS": "60000",
             "EXTENSION": "ndjson",
             "TEST": "false",
+            "RABBITMQ_PROTOCOL": "amqps",
             "RABBITMQ_HOST": "rabbitmq.default.svc.cluster.local:5672",
             "RABBITMQ_USER": "peter",
             "SWITCH_BRANCH": "master",
diff --git a/examples/spouts/go-rabbitmq-spout/source/main.go b/examples/spouts/go-rabbitmq-spout/source/main.go
index d63b472e2..87a16484f 100644
--- a/examples/spouts/go-rabbitmq-spout/source/main.go
+++ b/examples/spouts/go-rabbitmq-spout/source/main.go
@@ -311,7 +311,12 @@ func dialRabbit() (*amqp.Connection, error) {
                return nil, fmt.Errorf("RABBITMQ_PASSWORD environment variable not set")
        }

-       return amqp.Dial("amqp://" + rabbitUser + ":" + rabbitPassword + "@" + rabbitURL)
+       rabbitProtocol := os.Getenv("RABBITMQ_PROTOCOL")
+       if rabbitProtocol == "" {
+               rabbitProtocol = "amqp"
+       }
+
+       return amqp.Dial(rabbitProtocol + "://" + rabbitUser + ":" + rabbitPassword + "@" + rabbitURL)
 }

 func main() { 

```